include/kernel.h

```diff
@@ -95,7 +95,7 @@ typedef struct {
 #endif
 
 #ifdef CONFIG_OBJECT_TRACING
-#define _OBJECT_TRACING_NEXT_PTR(type) struct type *__next
+#define _OBJECT_TRACING_NEXT_PTR(type) struct type *__next;
```

Comment:
I'm not sure what happens here. You add a semicolon here to remove it in bunch of places and get the same result?

Reply:
yeah, this commit makes no sense at all for me

Reply:
> It seems it's to avoid some warnings of dangling semicolons for the condition when the macro expands to nothing. 

Ah, d'oh. Too bad that we have 2 threads discussing the same line of code...